web/pdf_scripting_manager.js

```diff
@@ -305,6 +305,27 @@ class PDFScriptingManager {
           }
           this._pdfViewer.currentScaleValue = value;
           break;
+        case "SaveAs":
+          this._eventBus.dispatch("save", { source: this });
+          break;
+        case "FirstPage":
+          this._eventBus.dispatch("firstpage", { source: this });
```

Comment:
Events should only be used for things that the `BaseViewer` doesn't *directly* support, and which require the full viewer (such as e.g. saving).

For everything else, please use direct `this._pdfViewer` calls wherever possible to avoid breaking the standalone viewer components unnecessarily; note https://github.com/mozilla/pdf.js/tree/master/examples/components

Hence, for the FirstPage, LastPage, NextPage, and PrevPage event specifically: please revert to the previous version of this PR (with my two comments addressed).